src/Context.tsx

```diff
@@ -7,7 +7,7 @@ const PuzzleContext = createContext<PuzzlesContextInterface>(null!);
 const PuzzleProvider: React.FC = ({children}) => {
```

Comment:
Removing this `React.FC` causes some problems that I'm not sure how to solve. I'm going to look at this tomorrow and see if I can provide some insight :)

Reply:
So I found this [stackoverflow](https://stackoverflow.com/questions/59988667/typescript-react-fcprops-confusion) that illuminated some stuff for me. Typing something as `React.FC` provides an implicit definition of `children`. This can actually be a big problem if your children aren't components. In your case it doesn't matter. I found an example of typing a provider in our codebase and following that example this file would look more like this:

<img width="448" alt="Screen Shot 2022-02-10 at 11 02 30 AM" src="https://user-images.githubusercontent.com/69563078/153468662-3599ab99-d547-4345-a39f-f07a649fca65.png">
 
 The argument for doing it this way is you can be more explicit about typing the children of this function. If you hover over ReactNode you see that it allows for a lot of different types
 
<img width="953" alt="Screen Shot 2022-02-10 at 11 03 16 AM" src="https://user-images.githubusercontent.com/69563078/153468808-b1be1959-453c-48e0-89b5-5e23c4d00da2.png">

So you could get even more explicit if you wanted to, but `ReactNode` feels like more than enough. The takeaway here is that TypeScript has a tremendous amount of layers and complexity, but at this stage this isn't something you should worry about too much, nor let it turn you off of TypeScript in the future. These aren't what I would call super important parts of the language.

Reply:
This is super interesting, thanks so much for looking into it! Made a note of it for best practice moving forward.